xmtp_mls/src/api/identity.rs

```diff
@@ -251,135 +241,4 @@ mod tests {
-            EncryptedMessageStore::new_unencrypted(StorageOption::Persistent(tmpdb)).unwrap();
-
-        let keypair = SignatureKeyPair::new(SignatureScheme::ED25519).unwrap();
-        let credential = Credential::new(CredentialType::Basic, rand_vec());
-        let stored: StoredIdentity = (&Identity {
-            inbox_id: inbox_id.clone(),
-            installation_keys: keypair,
-            credential,
-            signature_request: None,
-        })
-            .into();
-
-        stored.store(&store.conn().unwrap()).unwrap();
-
-        mock_api.expect_get_inbox_ids().returning(move |_| {
-            let wrong_address = "wrong".to_string();
-            Ok(GetInboxIdsResponse {
-                responses: vec![GetInboxIdsResponseItem {
-                    address: wrong_address,
-                    inbox_id: Some(inbox_id.clone()),
-                }],
-            })
-        });
-
-        let wrapper = ApiClientWrapper::new(mock_api, Retry::default());
-
-        let identity = IdentityStrategy::CreateIfNotFound(address.clone(), None);
-        let err = identity
-            .initialize_identity(&wrapper, &store)
-            .await
-            .unwrap_err();
-
-        assert!(matches!(err, IdentityError::NoAssociatedInboxId(addr) if addr == address));
-    }
-
-    #[tokio::test]
-    async fn identity_should_fail_on_wrong_inbox_id() {
-        let mut mock_api = MockApiClient::new();
-
-        let network_address = rand_string();
-        let inbox_id = "inbox_id".to_string();
-
-        let tmpdb = tmp_path();
-        let store =
-            EncryptedMessageStore::new_unencrypted(StorageOption::Persistent(tmpdb)).unwrap();
-
-        let keypair = SignatureKeyPair::new(SignatureScheme::ED25519).unwrap();
-        let credential = Credential::new(CredentialType::Basic, rand_vec());
-        let stored: StoredIdentity = (&Identity {
-            inbox_id: inbox_id.clone(),
-            installation_keys: keypair,
-            credential,
-            signature_request: None,
-        })
-            .into();
-
-        stored.store(&store.conn().unwrap()).unwrap();
-
-        let network_address_clone = network_address.clone();
-        mock_api.expect_get_inbox_ids().returning(move |_| {
-            Ok(GetInboxIdsResponse {
-                responses: vec![GetInboxIdsResponseItem {
-                    address: network_address_clone.clone(),
-                    inbox_id: Some("other".into()),
-                }],
-            })
-        });
-
-        let wrapper = ApiClientWrapper::new(mock_api, Retry::default());
-
-        let identity = IdentityStrategy::CreateIfNotFound(network_address.clone(), None);
-        let err = identity
-            .initialize_identity(&wrapper, &store)
-            .await
-            .unwrap_err();
-
-        assert!(
-            matches!(err, IdentityError::InboxIdMismatch{ id, address, stored} if id == "other" && address == network_address && stored == inbox_id)
-        );
-    }
```

Comment:
deleted these because they don't work for new Identity, added a few more tests in the builder.rs.